src/segment.rs

```diff
@@ -59,6 +60,7 @@ impl Segment {
 
     #[inline]
     pub fn set_best_segment_time(&mut self, time: Time) {
+        self.comparisons.insert(BEST_SEGMENT_COMPARISON_NAME.into(), time);
```

Comment:
The live split code adds the best segment to the comparisons via the best segments comparison generator, however that's not implemented yet but you are indexing comparisons on best segments which is returning None causing the live delta to not work correctly